PowerEditor/src/WinControls/shortcut/RunMacroDlg.cpp

```diff
@@ -130,17 +130,62 @@ intptr_t CALLBACK RunMacroDlg::run_dlgProc(UINT message, WPARAM wParam, LPARAM l
 					return TRUE;
 
 				case IDOK:
+				{
 					if (::SendDlgItemMessage(_hSelf, IDC_MACRO_COMBO, CB_GETCOUNT, 0, 0) > 0)
 						::SendMessage(_hParent, WM_MACRODLGRUNMACRO, 0, 0);
 
 					return TRUE;
+				}
 
 				default:
-					if ((HIWORD(wParam) == CBN_SELCHANGE) && (LOWORD(wParam) == IDC_MACRO_COMBO))
+				{
+					switch (LOWORD(wParam))
 					{
-						_macroIndex = static_cast<int32_t>(::SendDlgItemMessage(_hSelf, IDC_MACRO_COMBO, CB_GETCURSEL, 0, 0));
-						return TRUE;
+						case IDC_MACRO_COMBO:
+						{
+							if (HIWORD(wParam) == CBN_SELCHANGE)
+							{
+								_macroIndex = static_cast<int32_t>(::SendDlgItemMessage(_hSelf, IDC_MACRO_COMBO, CB_GETCURSEL, 0, 0));
+								return TRUE;
+							}
+							return FALSE;
+						}
+
+						case IDC_M_RUN_TIMES:
+						{
+							switch (HIWORD(wParam))
+							{
+								case EN_KILLFOCUS:
+								{
+									const int times = ::GetDlgItemInt(_hSelf, IDC_M_RUN_TIMES, nullptr, FALSE);
+									if (times < 1)
+									{
+										::SetDlgItemInt(_hSelf, IDC_M_RUN_TIMES, 1, FALSE);
+										return TRUE;
+									}
+
+									return FALSE;
+								}
+
+								case EN_CHANGE:
+								{
+									_times = std::max<int>(::GetDlgItemInt(_hSelf, IDC_M_RUN_TIMES, nullptr, FALSE), 1);

```

Comment:
@ozone10,

Thank you for this improvement.

Can't `_times =` be executed in `case EN_KILLFOCUS:`?

Reply:
@Yaron10 
when you change value and press "enter", focus is not changed to different control

Reply:
@ozone10,

Very good! 👍 

And you don't want to implement `case EN_CHANGE:` only.
We've discussed this option before. :)
It's actually implemented somewhere in the Preference dialog.

Reply:
> It's actually implemented somewhere in the Preference dialog.

In Backup -> Backup in every --- seconds.

Reply:
Enabling/disabling the Run button on `case EN_CHANG` might be a good option.

Reply:
@Yaron10 
> Enabling/disabling the Run button on `case EN_CHANGE:` might be a good option.

this will make it more complex, 
e.g. Run button disabled, switch to Run until the EOF or click on other control, 
now I need to handle this in `case EN_KILLFOCUS:`

> In Backup -> Backup in every --- seconds.

I don't think I've touched code for this.